### PR TITLE
Fixes #1240 properly spread callback args

### LINKED ISSUE #1240: mysql2 does not properly return fields
**Describe the bug**
I'm trying to run the agent with mysql2 and I get errors when i do 
```
connection.query('select * from foo;', (err, result, fields) => {
});
```
Looking at the instrumentation it is being a bit naive and only returning err and results.

**Environment**

* **Operation system:** N/A
* **Node version:** N/A
* **Tracer version:** v0.30.0
* **Agent version:** N/A

### PR DESCRIPTION
**Pull request by bizob2828**
### What does this PR do?
This properly spreads the callback args to fix #1240 

### Motivation
I want mysql2 support and use fields in my code

### Plugin Checklist

- [x] Unit tests.

[1]: https://github.com/DataDog/dd-trace-js/blob/master/index.d.ts
[2]: https://github.com/DataDog/dd-trace-js/blob/master/docs/test.ts
[3]: https://github.com/DataDog/dd-trace-js/blob/master/docs/API.md
[4]: https://github.com/DataDog/dd-trace-js/blob/master/.circleci/config.yml
[5]: https://github.com/DataDog/dd-trace-js/blob/master/packages/dd-trace/src/plugins/index.js

### Additional Notes
I updated a test, but can't get tests to run locally.

### CODE REVIEW

**Comment by bizob2828** [comment]
I'm not sure if the checklist above is necessary for my change so I will defer to the maintainers of this agent

File: packages/datadog-plugin-mysql2/src/index.js [hunk 1]
```diff
@@ -57,11 +57,19 @@ function wrapExecute (tracer, config, execute) {
 }
 
 function wrapCallback (tracer, span, parent, done) {
-  return tracer.scope().bind((error, res) => {
-    span.addTags({ error })
+  return tracer.scope().bind((...args) => {
+    const [ err ] = args;
+    if (err) {
```
Comment (rochdev) [inline packages/datadog-plugin-mysql2/src/index.js:62]: No need to expand to this condition, the previous code can stay as is as it is simpler.

File: packages/datadog-plugin-mysql2/test/index.spec.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const plugin = require('../src')
 
 wrapIt()
 
-describe('Plugin', () => {
+describe.only('Plugin', () => {
```
Comment (rochdev) [inline packages/datadog-plugin-mysql2/test/index.spec.js:8]: `only` should be removed.

**Review by rochdev — CHANGES_REQUESTED** [review]
Thanks for the PR! A few small things and then LGTM.

File: packages/datadog-plugin-mysql2/src/index.js (hunk 1 above)
Comment (bizob2828) [reply packages/datadog-plugin-mysql2/src/index.js:62]: @rochdev sounds good. I was just trying to conform to how it was in mysql plugin but ill update!

File: packages/datadog-plugin-mysql2/test/index.spec.js (hunk 2 above)
Comment (bizob2828) [reply packages/datadog-plugin-mysql2/test/index.spec.js:8]: shit, my bad

**Review by rochdev — APPROVED** [review]

**Comment by bizob2828** [comment]
@rochdev curious on your release cadence. when would this bug fix get shipped?

**Comment by rochdev** [comment]
> when would this bug fix get shipped?
We're planning a release next Tuesday.

**Comment by bizob2828** [comment]
@rochdev also thank you for such a fast review/merge 🙏🏻

**Comment by rochdev** [comment]
Released in 0.31.0.

**Pull request merged**